internal/gh/rest.go

```diff
@@ -0,0 +1,219 @@
+package gh
+
+import (
+	"net/http"
+    "net/url"
+    "regexp"
+    "log/slog"
+    "encoding/json"
+    "io"
+)
+
+const baseURL = "https://api.github.com"
+
+type Result struct {
+    Entry string
+}
+
+//  Create an HTTP request using the parameters, token, method and http client.
+func createHttpRequest(client *http.Client, reqUrl, requestMethod string, requestBody io.ReadCloser, queryParameters, headerParameters map[string] string)(*http.Request, error) {
+
+    req, err := http.NewRequest(requestMethod, reqUrl, requestBody)
+    if err != nil {
+        return req, err
+    }
+
+    for key, value := range headerParameters{
+        req.Header.Set(key, value)
+    }
+    
+    query := req.URL.Query()
+
+    for key, value := range queryParameters{
+        query.Add(key, value)
+    }
+
+    req.URL.RawQuery = query.Encode()
+
+    return req, nil
+}
+
+//  Update an existing HTTP request using the parameters, token and method.
+func updateHttpRequest(reqUrl, requestMethod string, requestBody io.ReadCloser, queryParameters, headerParameters map[string] string, req *http.Request) error{
+
+    parsedUrl, err := url.Parse(reqUrl)
+    if err != nil {
+        slog.Default().Error("Error while parsing the URL", err)
+	    return err
+    }
+
+    query := parsedUrl.Query()
+    req.Method = requestMethod
+    req.Body = requestBody
+
+    for key, value := range headerParameters{
+        req.Header.Set(key, value)
+    }
+
+    for key, value := range queryParameters{
+        if query.Has(key){
+            query.Set(key, value)
+        } else {
+            query.Add(key, value)
+        }
+        
+    }
+
+    req.URL.RawQuery = query.Encode()
+
+    return nil
+}
+
+//  Entry point for the Force Push processing, setting the url, query paramenters and creating the client.
+func GetForcePushInfo(owner, repo, token, branch string) (int, error){
+
+    slog.Default().Info("Getting repo info - getForcePushInfo method")
+    repoActivityUrl := baseURL + "/repos/" + owner + "/" + repo + "/activity"
+
+    queryParameters := map[string]string{
+        "per_page" : "100",
+        "activity_type" : "force_push",
+        "ref" : branch,
+    }
+
+    headerParameters := map[string]string{
+        "Content-Type" : "application/json",
+        "Authorization": "Bearer "+token,
+    }
+
+    client := &http.Client{}
+
+    NumberForcePush, err := processForcePushRequest(client, repoActivityUrl, queryParameters, headerParameters) 
```

Comment:
lower case for local variables 
```suggestion
    numberForcePush, err := processForcePushRequest(client, repoActivityUrl, queryParameters, headerParameters) 
```